scripts/test-check-false-positives.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+cleanup() {
+  r=$?
+
+  # Dump stack logs
+  elastic-package stack dump -v --output "build/elastic-stack-dump/check-${PACKAGE_UNDER_TEST:-${PACKAGE_TEST_TYPE:-*}}"
+
+  # Take down the stack
+  elastic-package stack down -v
+
+  # Clean used resources
+  for d in test/packages/${PACKAGE_TEST_TYPE:-false_positives}/${PACKAGE_UNDER_TEST:-*}/; do
+    (
+      cd $d
+      elastic-package clean -v
+    )
+  done
+
+  # This is a false positive scenario and tests that the test case failure is a success scenario  
+  if [ "${PACKAGE_TEST_TYPE:-false_positives}" == "false_positives" ]; then
+    if [ $r == 1 ]; then
+      rm build/test-results/*.xml
+      exit 0
```

Comment:
Just wondering if this could be leveraged to check specifically which tests are failing.

It could happen that some other test is failing and not the expected one. Or another scenario it would be that more tests are failing than expected.

I see that the resulting xml is like this 

```xml
<?xml version="1.0" encoding="UTF-8"?>
<testsuites>
  <testsuite name="system" tests="1" failures="1">
    <!--test suite for system tests-->
    <testcase name="system test: pagination" classname="httpjson_false_positive_asserts.generic" time="28.841040253">
      <failure>observed hit count 4 did not match expected hit count 2</failure>
    </testcase>
  </testsuite>
</testsuites>
```

Maybe we could add another file next to the folder `test/packages/false_positives/httpjson_false_positive_asserts`, something like:
`test/packages/false_positives/httpjson_false_positive_asserts-expected_errors.xml` or similar

and compare somehow those two xmls.

There would be differences because I see there is a field `time="XX.XX"`. Could it be checked at least the failures lines ? Or a line by line comparison with regexes ? 

```xml
<?xml version="1.0" encoding="UTF-8"?>
<testsuites>
  <testsuite name="system" tests="1" failures="1">
    <!--test suite for system tests-->
    <testcase name="system test: pagination" classname="httpjson_false_positive_asserts.generic" time="*">
      <failure>observed hit count 4 did not match expected hit count 2</failure>
    </testcase>
  </testsuite>
</testsuites>
```

Just testing with the contents of the xml:
- a new file could be added that contains the expected errors:
    - it can contain regexes and double quotes must be escaped.
  ```
  <testcase name=\"system test: pagination\" classname=\"httpjson_false_positive_asserts.generic\" time=\".*\"> * <failure>observed hit count 4 did not match expected hit count 2</failure>
  ```
- Then in this `scripts/test-check-packages-false-positives` it could be added the comparison line by line to check for these errors

Reply:
In this commit, I pushed an approach to deal with this
https://github.com/elastic/elastic-package/pull/1333/commits/51e33f4d8e93fc18b7eae030911167fb0f150f55

Let me know what you think @jsoriano @bhapas

Reply:
That changeset is supposed to fail the build (https://buildkite.com/elastic/elastic-package/builds/1142), I will do a couple of tests more
- same number of errors, but different one
- expected errors match with the actual errors

Reply:
LGTM. Maybe at some point we should migrate these scripts to go, they are starting to have some complexity.

Reply:
First build with the expected output
- Error: There are unexpected errors in httpjson_false_positive_asserts

Reply:
Test Expected Error Scenarios:

- [ ]  Number of expected and actual errors are not equal
- [ ]  Number of expected and actual errors are equal but different contents in the actual errors than expected
- [ ]  The Expected and actual errors are same